Userland/Libraries/LibCpp/Preprocessor.cpp

```diff
@@ -16,47 +16,48 @@ Preprocessor::Preprocessor(const String& filename, const StringView& program)
     : m_filename(filename)
     , m_program(program)
 {
-    GenericLexer program_lexer { m_program };
-    for (;;) {
-        if (program_lexer.is_eof())
-            break;
-        auto line = program_lexer.consume_until('\n');
-        bool has_multiline = false;
-        while (line.ends_with('\\') && !program_lexer.is_eof()) {
-            auto continuation = program_lexer.consume_until('\n');
-            line = StringView { line.characters_without_null_termination(), line.length() + continuation.length() + 1 };
-            // Append an empty line to keep the line count correct.
-            m_lines.append({});
-            has_multiline = true;
-        }
-
-        if (has_multiline)
-            m_lines.last() = line;
-        else
-            m_lines.append(line);
-    }
 }
 
 Vector<Token> Preprocessor::process_and_lex()
 {
-    for (; m_line_index < m_lines.size(); ++m_line_index) {
-        auto& line = m_lines[m_line_index];
-
-        bool include_in_processed_text = false;
-        if (line.starts_with("#")) {
-            auto keyword = handle_preprocessor_line(line);
-            if (m_options.keep_include_statements && keyword == "include")
-                include_in_processed_text = true;
-        } else if (m_state == State::Normal) {
-            include_in_processed_text = true;
+    Lexer lexer { m_program };
+    lexer.set_ignore_whitespace(true);
+    auto tokens = lexer.lex();
```

Comment:
It would've been nice to get this as an iterable, but that can happen someday later (or when it becomes a bottleneck)

Reply:
Yeah good idea, i added `Lex::lex_iterable()`.
Some places still use the non-iterable version, converting them was not as trivial.